qemu_mode/build_qemu_support.sh

```diff
@@ -386,6 +386,19 @@ else
   make -C libqasan CC="$CROSS $CROSS_FLAGS" && echo "[+] libqasan ready"
 fi
 
+#### Hooking support
+if [ "$ENABLE_HOOKING" = "1" ];then
+  echo "[+] ENABLING HOOKING"
```

Comment:
is there a reason this is not on by default? performance? if there is no noticable impact it should be on by default. if it does not work when compiled static then check for that.

Reply:
> is there a reason this is not on by default? performance? if there is no noticable impact it should be on by default.
By default, it is not compiled because it depends on glib which, if not available, will hinder the entire afl-qemu-trace compilation
Even when compiled, the user can choose to ignore hooking by not including hooking_bridge plugin. If they do, two things can be expected which, along with the compilation roadblock mentioned before, made me make this module optional
1. Performance: each block translation needs to be hooked, even though only the "entry" point hook is used -- I have noticed some performance penalty when translation/execution is hooked -- I may experiment with translation hook removal after the "entry" block at a later stage
2. If afl-qemu-trace is used as standalone, say for one-off testing -- the hooking system would impede debugging as it hijacks GDB

> if it does not work when compiled static then check for that.
I have not checked for that, but it will not interfere with a statically compiled AFL++. However, it does not really fit the static compilation use case because the user hooks themselves are to be placed in a shared library at this time. @vanhauser-thc should I still verify and update build_qemu_support accordingly?